packages/aws-rfdk/lib/core/lib/exporting-log-group.ts

```diff
@@ -111,10 +113,25 @@ export class ExportingLogGroup extends Construct {
       threshold: 1,
     });
 
+    // Define log retention retry options to reduce the risk of the rate exceed error
+    // as the default create log group TPS is only 5. Make sure to set the timeout of log retention function
+    // to be greater than total retry time. That's because if the function that is used for a custom resource
+    // doesn't exit properly, it'd end up in retries and may take cloud formation an hour to realize that
+    // the custom resource failed.
     const logRetention = new LogRetention(this, 'LogRetention', {
       logGroupName: props.logGroupName,
       retention: retentionInDays,
+      logRetentionRetryOptions: {
+        base: Duration.millis(200),
+        maxRetries: 7,
+      },
     });
+    const logRetentionFunctionLogicalId = 'LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8a';
```

Comment:
Strictly speaking, I don't believe this is a [logical ID](https://docs.aws.amazon.com/cdk/v2/guide/identifiers.html#identifiers_logical_ids). The [API docs](https://docs.aws.amazon.com/cdk/api/v2/docs/constructs.Node.html#trywbrfindwbrchildid) for `Node.tryFindChild(...)` document the argument with:

>   Identifier of direct child

And I believe this refers to the [construct ID](https://docs.aws.amazon.com/cdk/v2/guide/identifiers.html#identifiers_construct_ids) which is a unique ID within the scope of its parent in the [construct tree](https://docs.aws.amazon.com/cdk/v2/guide/constructs.html#constructs_tree).

TLDR: Can we rename this variable to `logRetentionFunctionConstructId`?

Same feedback applies to the similar change in the `LogGroupFactory` class below.

Reply:
I referenced the variable name from https://github.com/aws/aws-cdk/blob/205e493e7bd6c5212f0ae374fdee28128ea49afe/packages/%40aws-cdk/aws-logs/lib/log-retention.ts#L122-L130. Actually I think this id may be both logical id and construct id. It's logical id because in my testing I created different stacks, but they all have the same logical id for the function in the output cloudformation template. Meanwhile, it's also construct id for LogRetentionFunction.

Reply:
Updated it to `logRetentionFunctionConstructId` to reduce confusion.

Reply:
Ah, I see their code uses similar naming. Oddly, in CDK's docs on [logical IDs](https://docs.aws.amazon.com/cdk/v2/guide/identifiers.html#identifiers_logical_ids):

>   Unique IDs serve as the logical identifiers

The [docs for unique IDs](https://docs.aws.amazon.com/cdk/v2/guide/identifiers.html#identifiers_unique_ids) say that a resource's logical ID is derived from its construct path:

>   AWS CDK generates the necessary unique identifiers by concatenating the elements of the path and adding an 8-digit hash

The [path docs](https://docs.aws.amazon.com/cdk/v2/guide/identifiers.html#identifiers_paths) state:

>   We refer to the collection of IDs from a given construct, its parent construct, its grandparent, and so on to the root of the construct tree, as a path.

The root of the construct tree is the app. The stack construct ID should be a part of the path, so based on my understanding of the docs, you should get different logical IDs for different stacks within the same app. \*shrug\*

Reply:
Yes, the LogRetention resource that I created in my TestStacks has a different logical ID per stack. The logical id is concatenation of construct ids of its parent construct, its grandparent etc. But the LogRetentionFunction resource has a fixed logical ID that is not related to any parent constructs. I'm not sure why, but it's interesting to know.